spring-integration-gcp/src/main/java/org/springframework/integration/gcp/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -85,4 +98,12 @@ protected void doStop() {
 
 		super.doStop();
 	}
+
+	public AckMode getAckMode() {
+		return this.ackMode;
+	}
+
+	public void setAckMode(AckMode ackMode) {
+		this.ackMode = ackMode;
```

Comment:
I think the condition would not evaluate to true in that scenario, without the `NPE`, but I added the guard anyways.